src/main/java/hu/oe/nik/szfmv/automatedcar/systemcomponents/PedalPosition.java

```diff
@@ -0,0 +1,273 @@
+    }
+
+    public void gasPedalUp(){
+        gasPedalSwitch = false;
+        gasPedalValueDecrease();;
+    }
+
+    private void gasPedalValueDecrease(){
+        Thread valueChangeThread = new Thread(()->{
+            int counter = 0;
+            while(!gasPedalSwitch && gasPedalValue>=0.0)
+            {
+
+                if(PedalValueValidRange(gasPedalValue-increaseNumber(counter))) {
+                    gasPedalValue -= increaseNumber(counter);
+                }
+                else{
+                    gasPedalValue=0;
+                }
+                virtualFunctionBus.toPowerTrainPacket.setGasPedalValue(gasPedalValue);
+                virtualFunctionBus.guiInputPacket.setGasPedalValue(gasPedalValue);
+                counter++;
+                Sleep();
+            }
+        });
+        valueChangeThread.start();
+    }
+
+    public void breakPedalDown() {
+        if (!breakPedalSwitch) {
+            breakPedalSwitch = true;
+            breakPedalValueIncrease();
+        }
+    }
+
+    private void breakPedalValueIncrease(){
+        Thread valueChangeThread = new Thread(()->{
+            int counter = 0;
+            while(breakPedalSwitch && breakPedalValue<100.0)
+            {
+
+                if(PedalValueValidRange(breakPedalValue+increaseNumber(counter))) {
+                    breakPedalValue += increaseNumber(counter);
+                }
+                else{
+                    breakPedalValue=100;
+                }
+                virtualFunctionBus.toPowerTrainPacket.setBreakPedalValue(breakPedalValue);
+                virtualFunctionBus.guiInputPacket.setBreakPedalValue(breakPedalValue);
+                counter++;
+                Sleep();
+            }
+        });
+        valueChangeThread.start();
+    }
+
+    public void breakPedalUp(){
+        breakPedalSwitch = false;
+        breakPedalValueDecrease();
+    }
+
+    private void breakPedalValueDecrease(){
+        Thread valueChangeThread = new Thread(()->{
+            int counter = 0;
+            while(!breakPedalSwitch && breakPedalValue>0.0)
+            {
+
+                if(PedalValueValidRange(breakPedalValue-increaseNumber(counter))) {
+                    breakPedalValue -= increaseNumber(counter);
+                }
+                else{
+                    breakPedalValue=0;
+                }
+                virtualFunctionBus.toPowerTrainPacket.setBreakPedalValue(breakPedalValue);
+                virtualFunctionBus.guiInputPacket.setBreakPedalValue(breakPedalValue);
+                counter++;
+                Sleep();
+            }
+        });
+        valueChangeThread.start();
+    }
+
+    private int increaseNumber(int counter){
+        if(counter <5){
+            return 5;
+        }
+        else if(counter < 10){
+            return 10;
+        }
+        else{
+            return 15;
+        }
+    }
+
+    private void Sleep(){
+        try {
+            Thread.sleep(100);
```

Comment:
- Name all functions with lower case, like `sleep()`.
- Why sleep 100 milliseconds? Packets are evaluated through the function bus loop always. If you change the value in a parallel way, while other parts are reading the packet, that might just lead to unnecessary complexity and bugs. So I'm asking, that are these threads necessary to work?